include/fields/DateTimeField.php

```diff
@@ -426,13 +426,14 @@ private static function sanitizeDate($value, $user)
 		}
 
 		$value = str_replace('T', ' ', $value);
-		list($date, $time) = explode(' ', $value);
+		list($date, $time) = array_pad(explode(' ', $value), 2, '');
 		if (!empty($date) && !in_array($time, ['AM', 'PM'])) {
-			$date = self::__convertToDBFormat($date, $user->date_format);
-			$value = $date;
-			if (!empty($time)) {
-				$value .= ' ' . rtrim($time);
+			if (empty($time)) {
+				$date = self::__convertToDBFormat($date, $user->date_format);
+			} else {
+				$date = self::__convertToDBFormat($date.' ' . rtrim($time), $user->date_format);
```

Comment:
You can not convert the date and time function __convertToDBFormat.
This should not be changed, it was correctly.
__convertToDBFormat converts only the date check what the function doing.